main.tf

```diff
@@ -13,6 +13,39 @@ module "label" {
   tags       = "${var.tags}"
 }
 
+resource "aws_s3_bucket" "cache_bucket" {
```

Comment:
Bucket should not be created if `cache_enabled` is false. Just use a count parameter.